irc/isupport.go

```diff
@@ -0,0 +1,66 @@
+package irc
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ISupportList holds a list of ISUPPORT tokens
+type ISupportList struct {
+	Tokens      map[string]*string
```

Comment:
ISUPPORT tokens can be presented in three separate ways, `"TOKEN"`, `"TOKEN="`, or `"TOKEN=value"`. This is a pointer because `nil` represents the first type of token where it cannot contain a value, `""` (empty string) represents a token like `"TOKEN="` where the value is empty, and an actual value represents the last type of token.

Currently, it's a pointer so it can represent all these types of tokens, but it can be ripped out and added back in later if we need to send out that first type?